platform/sync-installer/sync-installer-impl/src/main/java/org/codice/ddf/sync/installer/impl/SynchronizedInstallerImpl.java

```diff
@@ -0,0 +1,578 @@
+    }
+
+    waitForBundles(getRemainingTime(startTime, maxWaitTime));
+    LOGGER.info(
+        "Finished uninstalling features in {} ms", (System.currentTimeMillis() - startTime));
+  }
+
+  @Override
+  public void stopBundles(String symbolicName, String... additionalSymbolicNames) {
+    Set<String> toStop = toSet(symbolicName, additionalSymbolicNames);
+    String bundleNames = String.join(", ", toStop);
+    LOGGER.info("Stopping the following bundles:[{}]", bundleNames);
+    long startTime = System.currentTimeMillis();
+
+    for (Bundle bundle : bundleContext.getBundles()) {
+      if (toStop.contains(bundle.getSymbolicName())) {
+        try {
+          bundle.stop();
+        } catch (BundleException e) {
+          throw new SynchronizedInstallerException(
+              "Failed to stop bundle [" + bundle.getSymbolicName() + "]");
+        }
+      }
+    }
+
+    LOGGER.info("Finished stopping bundles in [{}] ms", (System.currentTimeMillis() - startTime));
+  }
+
+  @Override
+  public void startBundles(String symbolicName, String... additionalSymbolicNames) {
+    Set<String> toStart = toSet(symbolicName, additionalSymbolicNames);
+    String bundlesNames = String.join(", ", toStart);
+    LOGGER.info("Starting the following bundles:[{}]", bundlesNames);
+    long startTime = System.currentTimeMillis();
+
+    for (Bundle bundle : bundleContext.getBundles()) {
+      if (toStart.contains(bundle.getSymbolicName())) {
+        try {
+          bundle.start();
+        } catch (BundleException e) {
+          throw new SynchronizedInstallerException(
+              "Failed to start bundle [" + bundle.getSymbolicName() + "]");
+        }
+      }
+    }
+
+    LOGGER.info("Finished starting bundles in [{}] ms", (System.currentTimeMillis() - startTime));
+  }
+
+  @Override
+  public void waitForBundles(String... symbolicNames) throws InterruptedException {
+    waitForBundles(DEFAULT_MAX_BUNDLE_WAIT, symbolicNames);
+  }
+
+  @Override
+  public void waitForBundles(long maxWaitTime, String... symbolicNames)
+      throws InterruptedException {
+    Set<String> toWaitFor =
+        symbolicNames.length == 0
+            ? getAllBundleSymbolicNames()
+            : Arrays.stream(symbolicNames).collect(Collectors.toSet());
+
+    Callable<Boolean> areBundlesReady = () -> getUnavailableBundles(toWaitFor).isEmpty();
+
+    try {
+      wait(
+          areBundlesReady,
+          maxWaitTime,
+          DEFAULT_POLLING_INTERVAL,
+          "Failed waiting for bundles to reach Active state.");
+    } catch (SynchronizedInstallerTimeoutException e) {
+      printBundleInformation(LOGGER::error, getUnavailableBundles(toWaitFor));
+      throw e;
+    }
+  }
+
+  private Set<String> getAllBundleSymbolicNames() {
+    return Arrays.stream(bundleContext.getBundles())
+        .map(Bundle::getSymbolicName)
+        .collect(Collectors.toSet());
+  }
+
+  @VisibleForTesting
+  protected void wait(
```

Comment:
✏️ We usually use package-private for methods left visible for testing.

Reply:
This is being overridden in tests. From what I've read, groovy cannot override private methods.

Reply:
package private not private and yes those can be overridden.